File: longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int count[100] = {0};
        for(int i=0;i<s.length();i++)
            count[s[i]-'A']++;
        int one = 0, res = 0;
        for(int i=0;i<90;i++)
        {
            res += count[i] - (count[i]%2);
            if(count[i])cout << (char)(i+'A') << " " << count[i] << endl;
            if(count[i]%2)one = 1;
        }
        return res+one;
    }
};

File: Longest Palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        boolean odd  = false;
        int[] cnt = new int[100];
        char[] ch = s.toCharArray();
        for(int i=0;i<ch.length;i++)
            cnt[ch[i]-'A']++;
        int res = 0;
        for(int i=0;i<100;i++)
        {
            if(cnt[i] % 2 == 1)
            {
                odd = true;
                cnt[i]--;
            }
            res += cnt[i];
        }
        return res + (odd?1:0);
    }
}
